specs/experimental/gov-token.md

```diff
@@ -0,0 +1,98 @@
+contract to consume the hooks and update its delegation and checkpoint mappings accordingly.
+
+If the call to the `GovernanceDelegation`'s `afterTokenTransfer` function fails, the token transfer MUST revert. This ensures
+that the `GovernanceDelegation` contract remains in sync with the `GovernanceToken`. Otherwise, the `GovernanceToken` could
+be left in an inconsistent state relative to the `GovernanceDelegation` contract, such as when a token transfer is successful
+but the delegation state is not updated.
+
+All delegation-related state, including the `delegates`, `checkpoints`, and `numCheckpoints` mappings, is gradually
+shifted from the `GovernanceToken` to the `GovernanceDelegation` contract through transactions that call the
+`GovernanceDelegation`'s hook (e.g. transfers). In the hook, the `GovernanceDelegation` contract should check if the `to`
+and `from` addresses have been migrated. If an address hasn't been migrated, the `GovernanceDelegation` contract should
+write the data from the `GovernanceToken`'s mappings for that address to its own state. When reading delegation data in the
+`GovernanceDelegation` contract for a delegator that hasn't been migrated, the `GovernanceDelegation` should pull the data
+from the `GovernanceToken`'s state.
+
+For backwards compatibility, the getter methods in the `GovernanceToken` MUST check if the data of a given address has been
+migrated or not. If the data has been migrated, the `GovernanceToken` MUST forward the call to the `GovernanceDelegation`
+contract. Otherwise, the `GovernanceToken` MUST read from its state.
+
+The `delegate` and `delegateBySig` functions in the `GovernanceToken` MUST forward the calls to the `GovernanceDelegation`
+contract, which implements the required delegation logic.
+
+### Token Minting
+
+`GovernanceToken` MUST have a `mint(address,uint256)` function with external visibility that allows the contract owner
+to mint an arbitrary number of new tokens to a specific address. This function MUST only be called by the contract
+owner, the `MintManager`, as enforced by the `onlyOwner` modifier inherited from the `Ownable` contract. When tokens
+are minted, the voting power of the recipient address MUST be updated accordingly in the `GovernanceDelegation` contract
+via the `afterTokenTransfer` hook. The total token supply is capped to `2^208 - 1` to prevent overflow risks in the voting
+system. If the total supply exceeds this limit, `_mint(address,uint256)`, as inherited from `ERC20Votes`, MUST revert.
+
+### Token Burning
+
+The contract MUST allow token holders to burn their own tokens using the inherited `burn(uint256)` or
+`burnFrom(address,uint256)` functions inherited from `ERC20Burnable`. When tokens are burned, the total supply and the
+holder's voting power MUST be reduced accordingly in the `GovernanceDelegation` contract via the `afterTokenTransfer` hook.
```

Comment:
and a new total supply checkpoint MUST be written (for `_totalSupplyCheckpoints` array) with updated for the new total supply (same for minting but the increased total supply)